sklearn/preprocessing/data.py

```diff
@@ -1719,10 +1726,15 @@ def _transform_selected(X, transform, selected="all", copy=True):
         X_sel = transform(X[:, ind[sel]])
         X_not_sel = X[:, ind[not_sel]]
 
-        if sparse.issparse(X_sel) or sparse.issparse(X_not_sel):
-            return sparse.hstack((X_sel, X_not_sel))
+        if order:
+            # As of now, X is expected to be dense array
```

Comment:
During that time, I was not sure how to handle sparse matrices with maintaining order of the selected features. Will update the method.